src/render_cmd.rs

```diff
@@ -151,8 +152,21 @@ impl RenderCmd {
 
             let mut cmd = base_cmd.clone();
 
+            // check if "--filter is set and filter the deployment values."
+            // return type will be a vector
+            let filtered_values;
```

Comment:
This paradigm has first-class rust support like so:

```rust
let filtered_values = match &self.opts.filter {
  Some(opts) => self.filter_deployment(opts.clone(), d.values.clone()),
  None => d.values.clone();
}
```

Reply:
Phew 😄 . Thanks